src/merkletree/index.js

```diff
@@ -0,0 +1,134 @@
+const { fork } = require("child_process")
+
+const {
+    BUILD_TREE,
+    SUCCESS,
+    ERROR,
+    deserialiseTree,
+} = require("./worker")
+
+const MerkleTree = require("./api")
+
+function wrapErrorObject(err, message) {
+    let newError
+    if (err instanceof Error) {
+        newError = Object.create(err)
+    } else {
+        newError = new Error()
+    }
+    if (err.stack) {
+        // copy stack
+        newError.stack = err.stack
+    }
+    // prefix with new message
+    newError.message = [message, err.message].filter(Boolean).join("\n")
+    return newError
+}
+
+class Worker {
+    async send(message) {
+        return new Promise((resolve, reject) => {
+            let result
+            if (this.proc) {
+                this.proc.kill()
```

Comment:
what does send function do? It kills processes, forks new workers?

Reply:
Added docs and tweaked it a bit to be clearer. It's mainly about attaching the appropriate listeners and resolving the promise at the right time.